src/commonMain/kotlin/org/cru/godtools/tool/model/EventId.kt

```diff
@@ -24,4 +20,11 @@ class EventId internal constructor(val namespace: String? = null, val name: Stri
                 }
             }.orEmpty()
     }
+
+    override fun equals(other: Any?) = other is EventId &&
+        namespace == other.namespace &&
+        name.equals(other.name, true)
+
+    override fun hashCode() = (namespace?.hashCode() ?: 0) * 31 + name.lowercase().hashCode()
+    override fun toString() = if (namespace != null) "$namespace:$name" else name
```

Comment:
These are just extra functions that you created for this class right? Would you use them by doing something like ```EventId(name = it).hashCode()```?

Reply:
`equals`, `hashCode`, and `toString` are functions provided on all objects (default implementations are in the `Object`/`Any` base class that everything inherits from).

- `equals` is used to provide equality comparison between two objects, `a == b` or `a.equals(b)` would trigger that method.
- `hashCode` is used to generate a hash for the object, this hash should be the same between all equal values. This hashCode is used by `HashMap` and similar objects to quickly split multiple objects up into buckets to improve performance.
- `toString()` provide a string representation of this object